### ISSUE TITLE
dnsdist: feature to skip arbitrary EDNS options in hashing 

### ISSUE THREAD

**Issue by phonedph1** [PowerDNS/pdns#10585]
- Program: dnsdist
 - Issue type: Feature request

### Short description
Ignoring some other option codes beyond cookies can greatly improve packet cache hits.

### Usecase
We have some devices that send 65001/65074 but do not care about their contents. Ignoring these increases normal packet cache hits considerably. 

### Description
A more generic way to set which options you want to skip, possibly as options sent to the packet cache.

```
-      if (optionCode == EDNSOptionCode::COOKIE) {
+      if (optionCode == EDNSOptionCode::COOKIE || optionCode == 65001 || optionCode == 65074) {
```

**Comment by rgacogne** [comment]
Makes sense to me, I guess `PacketCache::canHashPacket` and `PacketCache::hashAfterQname` could take a `const std::unordered_set<uint16_t>& optionsToSkip` parameter instead of the single boolean. A quick and dirty test suggests that it would not be detrimental to performance.
